convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,109 @@
+    """
+    if os.path.exists(FIREWALLD_CONFIG_FILE):
+        contents = []
+        with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+            contents = [line.strip() for line in handler.readlines() if line]
+
+        # Contents list is empty for some reason, better to assume that there
+        # is no content in the file that was read.
+        if not contents:
+            return False
+
+        # If the config file has this option set to true/yes, then we need to
+        # return True to ask the user to change it to self.
+        if "CleanupModulesOnExit=yes" in contents or "CleanupModulesOnExit=true" in contents:
+            return True
+
+    return False
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id == "oracle" and (system_info.version.major == 8 and system_info.version.minor >= 8):
+            if systeminfo.is_systemd_managed_service_running("firewalld"):
+                if _check_for_modules_cleanup_config():
+                    self.set_result(
+                        level="ERROR",
+                        id="FIREWALLD_MODULESS_CLEANUP_ON_EXIT_CONFIG",
+                        title="Firewalld is set to cleanup modules after exit.",
+                        description="Firewalld running on Oracle Linux 8 can lead to a conversion failure.",
+                        diagnosis="We've detected that firewalld unit is running and that causes iptables and nftables failures on Oracle Linux 8 and under certain conditions it can lead to a conversion failure.",
+                        remediation=(
+                            "Set the option CleanupModulesOnExit in /etc/firewalld/firewalld.conf to no prior to running convert2rhel:\n"
+                            "1. sed -i -- 's/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g' /etc/firewalld/firewalld.conf\n"
+                            "You can change the option back to yes after the system reboot - that is after the system boots into the RHEL kernel."
+                        ),
+                    )
+                else:
+                    # Firewalld is running but the configuration for CleanupModulesOnExit was not set to true/yes
+                    self.add_message(
+                        level="WARNING",
+                        id="FIREWALLD_IS_RUNNING",
+                        title="Firewalld is running",
+                        description=(
+                            "We've detected that firewalld is running and we couldn't find check for the CleanupModulesOnExit configuration. "
+                            "This means that a reboot will be necessary after the conversion is done to reload the kernel modules and prevent firewalld from stop working."
```

Comment:
The default is yes so if the option is missing in the config, I expect that to have the same effect as if the option is set to yes.
I've just verified that removing the option does not lead to firewall complaining.
So, in the `_check_for_modules_cleanup_config` function, you should treat the missing option as if it's set to yes.